CTFd/events/__init__.py

```diff
@@ -13,6 +15,19 @@ def subscribe():
     @stream_with_context
     def gen():
         for event in current_app.events_manager.subscribe():
+            # Filter notifications based on user and team ID
+            if event.type == "notification":
+                if (
+                    event.data["user"] is not None
+                    and event.data["user"] != get_current_user().id
```

Comment:
get_current_user requires you to query the db. Not worthwhile if you only need the user's ID.